### ISSUE TITLE
Add typescript support for ui instead of javascript

### ISSUE THREAD

**Issue by kapoorp99** [FlowiseAI/Flowise#2770]
We propose adding TypeScript support for the UI components of the project to enhance code quality, improve developer experience, and leverage the benefits of static typing.
**Motivation:**
TypeScript is a superset of JavaScript that adds static typing, which helps catch errors early during development and provides better tooling support. By adopting TypeScript for the UI components, we can:

Improve code readability and maintainability.
Enable better collaboration among developers through explicit type definitions.
Reduce runtime errors by catching type-related issues at compile time.
Take advantage of the rich ecosystem of TypeScript tools and libraries.

**Comment by HenryHengZJ** [comment]
Would love to switch to Typescript too, its quite an effort 😄

**Comment by kapoorp99** [comment]
I Will share the major pr but i have analyzed that whole repo needs to be changed, because the pipelines as well break if we start from the current repo status
@HenryHengZJ

**Comment by HenryHengZJ** [comment]
@kapoorp99 yeah it is a big refractor change, do you have something already in work? feel free to add me to your repo

**Comment by kapoorp99** [comment]
okk sure

**Comment by manudeli** [comment]
I wanna do this step by step
